portal-api/src/v1/users/validation/index.js

```diff
@@ -43,28 +55,65 @@ const updateWithoutCurrentPasswordRules = [
   usernameAndEmailMustMatch,
   emailMustBeValidEmailAddress,
   emailMustBeUnique,
+  adminFromUkefOnly,
 ];
 
 const updateWithCurrentPasswordRules = [...updateWithoutCurrentPasswordRules, currentPasswordMustMatch];
 
-const applyRules = async (ruleset, existingUser, candidateChange) => {
+/**
+ * Applies a set of validation rules to a candidate change.
+ *
+ * @param {Array<Function>} ruleset - An array of validation rule functions to apply.
+ * @param {Object|null} existingUser - The existing user data, or null if creating a new user.
+ * @param {Object} candidateChange - The candidate change data to validate.
+ * @returns {Promise<Array<string>>} A promise that resolves to an array of validation error messages.
+ */
+const applyRules = async (ruleset, candidateChange, existingUser = null) => {
   let errors = [];
+
   for (const rule of ruleset) {
     const result = await rule(existingUser, candidateChange);
 
     if (result.length) {
+      // We concat here as rules return an array
       errors = errors.concat(result);
-    } // We concat here as rules return an array
+    }
   }
 
   return errors;
 };
 
-const applyCreateRules = async (candidateChange) => await applyRules(createRules, null, candidateChange);
+/**
+ * Applies the creation rules to the candidate change.
+ *
+ * @param {Object} candidateChange - The candidate change data to validate.
+ * @returns {Promise<Array<string>>} A promise that resolves to an array of validation error messages.
+ */
+const applyCreateRules = async (candidateChange) => await applyRules(createRules, candidateChange);
 
+/**
+ * Applies the update rules to the candidate change.
+ *
+ * Selects the appropriate set of rules based on whether the current password is provided.
+ *
+ * @param {Object} existingUser - The existing user data.
+ * @param {Object} candidateChange - The candidate change data to validate.
+ * @returns {Promise<Array<string>>} A promise that resolves to an array of validation error messages.
+ */
 const applyUpdateRules = async (existingUser, candidateChange) => {
   const rules = !candidateChange.currentPassword ? updateWithoutCurrentPasswordRules : updateWithCurrentPasswordRules;
-  return await applyRules(rules, existingUser, candidateChange);
+  /**
+   * Below is to satisfy any email address pertinent rules.
+   * The property has been set to `emailAddress` instead of using
+   * existing property `username` and `email` for current
+   * rules processing.
```

Comment:
I'm not understanding why we want to use `emailAddress` instead of `email`?

Reply:
Apologies for the confusion explanation below.

Since `email` attribute is used for creation, using the same attribute for update causes various update rules to throw error. Using a new attribute purely for update ruleset.